dxr/app.py

```diff
@@ -269,45 +303,29 @@ def _browse_folder(tree, path, config):
             for f in files_and_folders])
 
 
-def _browse_file(tree, path, config):
-    """Return a rendered page displaying a source file.
-
-    If there is no such file, raise NotFound.
+def skim_file(skimmers, num_lines):
+    """Skim contents with all the skimmers, returning the things we need to
+    make a template. Compare to dxr.build.index_file
 
+    :arg skimmers: iterable of FileToSkim objects
+    :arg num_lines: the number of lines in the file being skimmed
     """
-    def sidebar_links(sections):
-        """Return data structure to build nav sidebar from. ::
-
-            [('Section Name', [{'icon': ..., 'title': ..., 'href': ...}])]
-
-        """
-        # Sort by order, resolving ties by section name:
-        return sorted(sections, key=lambda section: (section['order'],
-                                                     section['heading']))
-
-    frozen = frozen_config(tree)
-
-    # Grab the FILE doc, just for the sidebar nav links:
-    files = filtered_query(
-        frozen['es_alias'],
-        FILE,
-        filter={'path': path},
-        size=1,
-        include=['links'])
-    if not files:
-        raise NotFound
-    links = files[0].get('links', [])
-
-    lines = filtered_query(
-        frozen['es_alias'],
-        LINE,
-        filter={'path': path},
-        sort=['number'],
-        size=1000000,
-        include=['content', 'tags', 'annotations'])
-
-    # Common template variables:
-    common = {
+    links, refses, regionses = [], [], []
+    annotations_by_line = [[] for _ in xrange(num_lines)]
+    for file_to_skim in skimmers:
```

Comment:
If you're going to call them skimmers (which is great), you might as well be consistent and say "for skimmer in skimmers".